File: quality_code_release_v1.0/dct_dft_feat.m
function [feats score2_vec fr_s fr_e] = dct_dft_feat(boxes, fname_annot, len1, n_seg, feat_type)

cents_x = (boxes(:, 1:4:end-3)+boxes(:, 3:4:end-3))/2;
cents_y = (boxes(:, 2:4:end-3)+boxes(:, 4:4:end-3))/2;

[fr_s fr_e score2_vec] = read_annotation(fname_annot);

for i = 1:length(fr_s)
  frs = [fr_s(i):fr_e(i)];
  cents_x1 = cents_x(frs, :);
  cents_y1 = cents_y(frs, :);
  
  %%% normalize relative to the head location
  dist_feat = [];
  tmpx = bsxfun(@minus, cents_x1, cents_x1(:, 1));
  tmpy = bsxfun(@minus, cents_y1, cents_y1(:, 1));
  dist_feat = [dist_feat tmpx tmpy];
  
  %%% for each joint: collect absolute value in the frequency domain
  feat2 = [];
  r = length(frs)/n_seg;
  for j = 1:n_seg
    dist_feat1 = dist_feat(round((j-1)*r+1) : round(j*r), :);
    
    if isequal(feat_type, 'pose+DCT')
      feat1 = dct1(dist_feat1, len1);
    elseif isequal(feat_type, 'pose+DFT')
      feat1 = fft(dist_feat1);
      feat1 = feat1(1:len1, :);
    end
    feat2 = [feat2; abs(feat1)];
  end
  feats(:, i) = feat2(:);
end

File: dct_svr/dct_dft_feat.py
from math import *
from dct1 import *
from numpy import *

def dct_dft_feat(boxes, fname_annot, len1, n_seg):

    cents_x = (boxes[:, 0:-4:3] + boxes[:, 2:-4:3]) / 2
    cents_y = (boxes[:, 1:-4:3] + boxes[:, 3: -4:3]) / 2

    # read_annotation function
    fr_s,fr_e, score2_vec = read_annotation(fname_annot)

    feats = []

    for i in range(len(fr_s)):
        frs = arange(fr_s(i), fr_e(i))
        cents_x1 = cents_x[frs,:]
        cents_y1 = cents_y[frs,:]

        dist_feat = []
        # not super sure how to replace the bsxfun here
        tmpx = bsxfun(subtract, cents_x1, cents_x1[:, 0])
        tmpy = bsxfun(subtract, cents_y1, cents_y1[:, 0])
        dist_feat = [dist_feat, tmpx, tmpy]

        feat2 = []
        r = len(frs) / n_seg
        for j in range(1, n_seg+1):
            dist_feat1 = dist_feat[round((j - 1) * r + 1): round(j * r),:]

        # feat1 and feat2 having an error here
        feat1 = dct1[dist_feat1, len1]
        feat2 = np.block([[feat2],[abs(feat1)])
    feats[:, i] = feat2.flatten()
